### PR TITLE
New test structures

### PR DESCRIPTION
**Pull request by natoverse**
Adds support for JSON in cross-platform fixtures, which allows rewriting tests that were inadvertently obscuring issues in the verbs. This migrates a handful of tests to JSON to address some of those issues, but a deeper review is needed later. This also removes some date and object parsing "cheats" in the tests since JSON naturally handles those types better, and it removes a couple of tests that were invalid but masked by said cheats.

### CODE REVIEW

File: python/datashaper/tests/test_schema.py [hunk 1]
```diff
@@ -25,42 +25,28 @@
...
         if file.endswith(".csv"):
-            path = Path(TABLE_STORE_PATH) / file
-            table = pd.read_csv(
-                path,
-                dtype_backend=pandas_dtype_backend,
-                engine=engine,
-            )
+            table = read_csv(path)
             dataframes[file.removesuffix(".csv")] = table
+        if file.endswith(".json") and file != "workflow.json":
```
Comment (darthtrevino) [inline python/datashaper/tests/test_schema.py:36]: We're doing a lot of `endswith` in this PR - maybe it's worth a filename extension utility in datashaper/core or datashaper/tables?

File: schema/fixtures/workflow/verbs/destructure/plain_object_remove_column_no_keys/expected.json [hunk 2]
```diff
@@ -0,0 +1,66 @@
+{
+    "ID": [
```
Comment (darthtrevino) [inline schema/fixtures/workflow/verbs/destructure/plain_object_remove_column_no_keys/expected.json:2]: the columnar json is a little ugly, but that's more of an aside than anything

**Review by darthtrevino — APPROVED** [review]

File: python/datashaper/tests/test_schema.py [hunk 3]
```diff
@@ -25,42 +25,28 @@
 
 
 @cache
-def load_inputs(pandas_dtype_backend: PandasDtypeBackend, engine: str):
```
Comment (darthtrevino) [inline python/datashaper/tests/test_schema.py:28]: So the issue here is that the tests aren't compatible with the Pyarrow backend anymore right?

Comment (natoverse) [reply python/datashaper/tests/test_schema.py:28]: Correct - pyarrow requires jsonl, but regular engine (ujson) doesn't support it. I had to choose one, and default pandas made sense until pyarrow is 100% supported as an engine, and we may end up just using that for polars anyway.

File: schema/fixtures/workflow/verbs/destructure/plain_object_remove_column_no_keys/expected.json (hunk 2 above)
Comment (natoverse) [reply schema/fixtures/workflow/verbs/destructure/plain_object_remove_column_no_keys/expected.json:2]: Agreed, but it is the standard for pandas and built-in to arquero so it kept the file loading minimal

File: python/datashaper/tests/test_schema.py (hunk 1 above)
Comment (natoverse) [reply python/datashaper/tests/test_schema.py:36]: Yeah, good point - we _do_ have that in the js impl, py is just way behind on file loading stuff

**Pull request merged**